File: 11_sorts/03_selection_sort/selection_sort.js
function selection_sort(arr) {

	for (var i = 0; i < arr.length - 1; i++) {
		var select = i;
		for (var j = i + 1; j < arr.length; j++) {
			if (arr[j] < arr[select]) {
				select = j;
			}
		}
		if (select != i) {
			var temp = arr[i];
			arr[i] = arr[select];
			arr[select] = temp;
		}
	}
}

var arr = [11, 3, 4, 2, 8, -3, 6, 12, -2, 10, 0, 5];
selection_sort(arr);
console.log(arr);

File: 11_sorts/03_selection_sort/selection_sort.py
def selection_sort(arr):
	array_length = len(arr)

	for i in range(array_length):
		select = i
		for j in range(i + 1, array_length):
			if arr[j] < arr[select]:
				select = j

		# swapping value with the selection
		arr[i], arr[select] = arr[select], arr[i]


arr = [11, 3, 4, 2, 8, -3, 6, 12, -2, 10, 0, 5]
selection_sort(arr)
print(arr)
